### PR TITLE
Configurable naming scheme for downloaded files

### PR DESCRIPTION
**Pull request by AbdullahM0hamed**
Because_Not_All_Of_Us_Like_This_Naming_Scheme.

### CODE REVIEW

File: anime_downloader/util.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from anime_downloader import session
 from anime_downloader.sites import get_anime_class, helpers
 from anime_downloader.const import desktop_headers
+from anime_downloader.config import Config
```
Comment (Blatzar) [inline anime_downloader/util.py:19]: Importing config in utils causes circular import on generation of config, moving this to slugify works

**Comment by ArjixGamer** [comment]
dead pr

**Comment by MichaelDavidJohnson** [comment]
Close please.

**Comment by ArjixGamer** [comment]
no, dont close it, dont listen to iggy, he dumb

File: anime_downloader/util.py [hunk 2]
```diff
@@ -175,9 +175,12 @@ def get_json(url, params=None):
     return data
 
 
-def slugify(file_name):
-    file_name = str(file_name).strip().replace(' ', '_')
-    return re.sub(r'(?u)[^-\w.]', '', file_name)
+def slugify(file_name): 
+    from anime_downloader.config import Config
+    seperator = Config._CONFIG["dl"]["title_seperator"]
+    file_name = str(file_name).strip().replace(' ', seperator)
```
Comment (vn-ki) [inline anime_downloader/util.py:181]: Can you add a TODO to investigate a more robust solution?

**Comment by Blatzar** [comment]
Doesn't work on twist as it retuns names separated with '-' (eg boruto-naruto-next-generations) as file_name

**Comment by MichaelDavidJohnson** [comment]
Yeah if we are gonna do something like remove all _ we should be able to user define if it's like a _ or a -, for future proofing. 
You could also argue it's outside the scope of the project and a batch script could be run after the download to bulk rename, but that's just me :P

**Comment by vn-ki** [comment]
I actually liked this pr. :P 

I understand everyone might not like `_`.

**Pull request closed without merging**